File: 1188/1188.cs
public class Solution {
    public int DesignBoundedBlockingQueue(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(DesignBoundedBlockingQueue(root.left), DesignBoundedBlockingQueue(root.right));
    }
}

File: 1188/1188.go
func designBoundedBlockingQueue(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := designBoundedBlockingQueue(root.Left)
	r := designBoundedBlockingQueue(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}